### ISSUE TITLE
Sentence tokenization with PunktSentenceTokenizer

### ISSUE THREAD

**Issue by furlat** [abstractqqq/polars_ds_extension#56]
https://www.nltk.org/api/nltk.tokenize.PunktSentenceTokenizer.html

I am working with llms and store input/outputs in polars (here for adding an embeddings column and using polars for hybrid filters/vector search). I often want as a processing step to tokenize into sentences but nltk is very slow, this seems like an interesting algorithm for a rust implementation that can have a lot of uses from multi sentence string column to list[str] column.

**Comment by abstractqqq** [comment]
This will be a great addition to the current string stuff and indeed there is a rust crate that does punkt sentence tokenization.

Only 1 problem. One of my goals is to keep the package as slim as possible (while keeping the performance). Punkt will require you to download predefined rule sets and will make the package much bigger. I can go about this like NLTK by asking the user to download these rules. I don't know how to do that yet. Another thing is that the Rust punkt crate is quite old and I am slightly worried about the maintenance.

Is this a high priority issue? Currently I don't think I will work on this in the near future. Hugging face seems to have a Rust-tokenizer crate which has a Python package that does some fancy tokenizations. Maybe that's helpful?

**Comment by furlat** [comment]
Thanks (again) for the great response, just to clarify I am loving the design package so I am giving suggestions of things that would fit well. Nothing I am asking is a real blocker for me since I am opening issues for algos I already have in my python stack and I think are worth to be "primitives" for dataframe pipelines.

I agree Punkt is low priority as I can always pre-process my strings slowly and keep them in a parquet for lazy processing. Something I observed is that going from utf8 to list[utf8] (after nltk punkt) is crazy expensive in terms of memory and was thinking that it would be better to do it on the spot only when needed to avoid storing the inflated parquet file. 

I was literally running punkt  (and waiting 40 mins for 3M rows) while "studying" this package and I felt like it would be a good fit, I did check the Rust Punkt crate but is indeed old :/  

I will post more suggestions in the next days and next week I will throughly test and benchmark the knn module on some text embeddings. 

Thanks so much for reading and considering my suggestions!
